api/src/main/java/org/pmiops/workbench/api/OfflineClusterController.java

```diff
@@ -0,0 +1,139 @@
+package org.pmiops.workbench.api;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.inject.Provider;
+
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.exceptions.ExceptionUtils;
+import org.pmiops.workbench.exceptions.ServerErrorException;
+import org.pmiops.workbench.model.CheckClustersResponse;
+import org.pmiops.workbench.notebooks.ApiException;
+import org.pmiops.workbench.notebooks.NotebooksConfig;
+import org.pmiops.workbench.notebooks.api.ClusterApi;
+import org.pmiops.workbench.notebooks.model.Cluster;
+import org.pmiops.workbench.notebooks.model.ClusterStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+
+/**
+ * Offline cluster API. Handles cronjobs for cleanup and upgrade of older
```

Comment:
So this says:

https://stackoverflow.com/questions/22859423/combining-google-app-engine-cron-job-and-task-is-starting-google-tasks-as-cron

That there's a ten minute timeout for handling these requests. That should be sufficient unless we get a whole lot of clusters or deletes are super-slow... but either way I'd document that here. 

If you wanted to avoid that timeout you could run this as a background task (but lose the ability to return the info in the response) or I guess we could look into having backend instances for workbench (which we might need at some stage...) Probably not necessary.

Reply:
Added to comment. I don't think we're going to come close to this, especially if we run this cron frequently.